[CIRCSTORE-127] Request update database trigger (#164)
folio-org/mod-circulation-storage: README.MD

See [configuration](https://dev.folio.org/download/artifacts) for repository access,
and the [Docker image](https://hub.docker.com/r/folioorg/mod-circulation-storage/).

## Design notes

> ### Request status transition

Reviewer: Please can some context be added to this, to refer to the report in mod-circulation and the reason why this has been implemented here, rather than in mod-circulation

There is database trigger for request update. It works in the following way:

> Given request with status 'Open - Awaiting pickup'.

Reviewer: ```suggestion
Given a request with status 'Open - Awaiting pickup'
```

> The request is updated and new status is 'Closed - Pickup expired' or 'Closed - Cancelled'.

Reviewer: ```suggestion
When the request is updated and new status is 'Closed - Pickup expired' or 'Closed - Cancelled'
```

> Than an update trigger adds new property 'awaitingPickupRequestClosedDate' to the request jsonb.

Reviewer: ```suggestion
Then an update trigger adds the property 'awaitingPickupRequestClosedDate' to the request jsonb
```
